packages/dds/tree/src/forest/cursor.ts

```diff
@@ -25,23 +35,39 @@ export const enum TreeNavigationResult {
  * (and likely via returning a sub-interface with documentation on the subject).
  *
  * TODO: Needs a way to efficiently clone Cursor so patterns like lazy tree reification can be implemented efficiently.
+ *
+ * TODO: add optional fast path APIs for more efficient handling when supported by underlying format and reader.
+ * Leverage "chunks" and "shape" for this, and skip to next chunk with seek (chunk length).
+ * Default chunks of size 1, and "node" shape?
  */
 export interface ITreeCursor {
     /** Select the child located at the given key and index. */
     down(key: FieldKey, index: number): TreeNavigationResult;
 
+    /**
+     * Moves `offset` entries in the field.
+     * May move less if Pending or NotFound.
+     * In this case the distance moved is returned, and may be less than `offset`.
+     * Iff `ok` then `moved` will equal `offset`.
+     */
+    seek(offset: number): { result: TreeNavigationResult; moved: number; };
```

Comment:
Oh, I like this a lot.  It replaces/generalizes the next/prev I had proposed earlier.

So far, the result of failed navigations has been that the cursor remains in its initial state (i.e., doesn't move), and 'Pending' is a failed navigation that can be retried once the pending chunks are loaded.

...but I'm open to trying other approaches.  Every option seems klunky in some way.

Reply:
ya: I'm not supper happy with it, but its seems good enough for now. We can get a cleaner API with copy on write navigation, but that defeats a lot of the whole point of the cursor.